knora/dsplib/utils/xml_upload.py

```diff
@@ -910,94 +883,435 @@ def xml_upload(input_file: str, server: str, user: str, password: str, imgdir: s
+    have an identical time stamp.
+
+    Args:
+        input_file: the file name of the original XML file
+        id2iri_mapping: mapping of ids from the XML file to IRIs in DSP
+        timestamp_str: timestamp for log file identification
+
+    Returns:
+        None
+    """
+
+    id2iri_mapping_file = "id2iri_" + Path(input_file).stem + "_mapping_" + timestamp_str + ".json"
+    with open(id2iri_mapping_file, "w") as outfile:
+        print(f"The mapping of internal IDs to IRIs was written to {id2iri_mapping_file}")
+        outfile.write(json.dumps(id2iri_mapping))
+
+
+def write_stashed_xml_texts(
+    stashed_xml_texts: dict[XMLResource, dict[XMLProperty, dict[str, KnoraStandoffXml]]],
+    timestamp_str: str
+) -> None:
+    """
+    Write the stashed_xml_texts into a file. The timestamp must be created by the caller, so that different log files
+    can have an identical time stamp.
+
+    Args:
+        stashed_xml_texts: all xml texts that have been stashed
+        timestamp_str: timestamp for log file identification
+
+    Returns:
+        None
+    """
+
+    filename = f'stashed_text_properties_{timestamp_str}.txt'
+    print(f'There are stashed text properties that could not be reapplied to the resources they were stripped from. '
+          f'They were saved to {filename}')
+    with open(filename, 'a') as f:
+        f.write('Stashed text properties that could not be reapplied\n')
+        f.write('***************************************************\n')
+        f.write('During the xmlupload, some text properties had to be stashed away, because the salsah-links in their '
+                'XML text formed a circle. The xmlupload can only be done if these circles are broken up, by stashing '
+                'away some of the chain elements of the circle. \n'
+                'Some of the resources that have been stripped from some of their text properties have been created in '
+                'DSP, but the stashed text properties could not be reapplied to them, because the xmlupload was '
+                'unexpectedly interrupted. \n'
+                'This file is a list of all text properties that are now missing in DSP. The texts have been replaced '
+                'by a hash number that now stands in the text field in DSP. \n'
+                '(Not listed are the stripped resources that haven\'t been created in DSP yet.) \n')
+        for res, props in stashed_xml_texts.items():
+            f.write(f'\n{res.id}')
+            f.write('\n' + '=' * len(res.id))
+            for prop, stashed_texts in props.items():
+                if len(stashed_texts) > 0:
+                    f.write(f'\n{prop.name}')
+                    f.write('\n' + '-' * len(prop.name))
+                    for hash, standoff in stashed_texts.items():
+                        f.write(f'\ntext with hash {hash}:\n{str(standoff).strip()}\n')
```

Comment:
is there a reason to write this to a .txt file?
If this was written to e.g. a JSON file, it would be much easier to get the information out again, especially in the case of a very big upload.